### PR TITLE
fix(builder): js layout

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Relate to (and reverts) #4641

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Review by TheAlexLichter — DISMISSED** [review]

**Comment by pi0** [comment]
Thanks @clarkdo, @manniL for addressing this so fast :heart:

**Review by pi0 — DISMISSED** [review]

File: packages/builder/src/builder.js [hunk 1]
```diff
@@ -349,7 +349,7 @@ export default class Builder {
           }
           return
         }
-        if (!templateVars.layouts[name] && /\.vue$/.test(file)) {
+        if (!templateVars.layouts[name] || /\.vue$/.test(file)) {
```
Comment (pi0) [inline packages/builder/src/builder.js]: Can we sort [`layoutFiles`](https://github.com/nuxt/nuxt.js/blob/fc6fb7a75719fdb6b9c5f4fb375d9db340f26ff4/packages/builder/src/builder.js#L339) here to give `.vue` a higher priority or at least add a comment above this line to explain the reason of check? It seems a hidden workaround like this.

Comment (clarkdo) [reply packages/builder/src/builder.js]: I think comment is better, since sorting may introduce extra overhead

Comment (manniL) [reply packages/builder/src/builder.js]: As long as we make it *understandable* to ppl who read the code, why not ☺️

**Review by TheAlexLichter — APPROVED** [review]

**Review by pi0 — APPROVED** [review]

**Pull request merged**